src/index.js

```diff
@@ -45,3 +45,22 @@ exports.Queue = Plotly.Queue;
 
 // export d3 used in the bundle
 exports.d3 = require('d3');
+
+Plotly.register({
+    traces: {
+        bar: require('./traces/bar'),
```

Comment:
The trace modules have `_type` in them right now, but it's more their name - I just followed the names of the arguments for the `Plots.register` function. I'd be happy to go either way with this though - the object notation allows for easily extending functionality in the future, but it *is* a bit more verbose. Up to you guys.